e2e/libs/keywords/volume_keywords.py

```diff
@@ -1,34 +1,47 @@
-from pathlib import Path
-from volume import Volume
-from utility import Utility
-from node import Node
+from utility.utility import get_test_case_namespace, generate_volume_name
+from utility.utility import get_node, list_nodes
+from utility.utility import get_test_pod_running_node, get_test_pod_not_running_node
+from robot.libraries.BuiltIn import BuiltIn
 from node_exec import NodeExec
+from volume import Volume
+import logging
 
 class volume_keywords:
 
-    ROBOT_LIBRARY_SCOPE = 'TEST'
-
     def __init__(self):
-        Utility().init_k8s_api_client()
-
-
-    def set_test_name(self, test_name):
-        self.namespace = test_name.lower().replace(' ', '-')
-        self.node_exec = NodeExec(self.namespace)
+        #TODO
+        #test_name = BuiltIn().get_variable_value("${TEST NAME}")
+        self.node_exec = NodeExec("default")
         self.volume = Volume(self.node_exec)
 
 
     def create_volume(self, size, replica_count):
-        print('create_volume')
-        volume_name = Utility().generate_volume_name()
+        volume_name = generate_volume_name()
         self.volume.create(volume_name, size, replica_count)
+        logging.info(f'==> create volume {volume_name}')
         return volume_name
 
 
-    def attach_volume(self, volume_name, attached_node_index=0):
-        print('attach_volume')
-        node_name = Node().get_by_index(attached_node_index)
-        self.volume.attach(volume_name, node_name)
+    def attach_volume(self, volume_name):
+        attach_node = get_test_pod_not_running_node()
+        logging.info(f'==> attach volume {volume_name} to {attach_node}')
+        self.volume.attach(volume_name, attach_node)
+
+
+    def get_volume_attached_node(self, volume_name):
+        volume = self.volume.get(volume_name)
+        print(volume)
+        return volume['spec']['nodeID']
+        # return volume.controllers[0].hostId
+
+
+    def get_volume_not_attached_node(self, volume_name):
```

Comment:
It's not a random node. Since we only have 3 worker nodes, and one of them is volume node, another of them is test pod running node, so there's only one node left can be selected.

Reply:
Modified to `get_replica_node` for consistency.